src/aosm/azext_aosm/_configuration.py

```diff
@@ -473,6 +328,189 @@ def validate(self):
             )
 
 
+NFD_NAME = (
+    "The name of the existing Network Function Definition to deploy using this NSD"
+)
+NFD_VERSION = "The version of the existing Network Function Definition to base this NSD on.  This NSD will be able to deploy any NFDV with deployment parameters compatible with this version."
+NFD_LOCATION = "The region that the NFDV is published to."
+PUBLISHER_RESOURCE_GROUP = "The resource group that the publisher is hosted in."
+PUBLISHER_NAME = "The name of the publisher that this NFDV is published under."
+NFD_TYPE = "Type of Network Function. Valid values are 'cnf' or 'vnf'"
+MULTIPLE_INSTANCES = (
+    "Set to true or false.  Whether the NSD should allow arbitrary numbers of this "
+    "type of NF.  If set to false only a single instance will be allowed.  Only "
+    "supported on VNFs, must be set to false on CNFs."
+)
+
+
+@dataclass
+class NFDRETConfiguration:
+    publisher: str = PUBLISHER_NAME
+    publisher_resource_group: str = PUBLISHER_RESOURCE_GROUP
+    name: str = NFD_NAME
```

Comment:
Across the AOSM team, we are inconsistent about "groups".  Some want to hide the fact that we've mis-named things as Groups. Some want to be consistent with the HybridNetwork API.  Personally, I'd call this group name, given that's what the resource ultimately is.

Reply:
This is a pain to rename (VS code hangs renaming the symbol and grepping for name turns out to be a bad idea) and this isn't the only place that we would want to change to be consistent, so I'm going to leave this for now.